### DISCUSSION TITLE
How to combine Databricks Workflows from multiple projects?

### DISCUSSION THREAD

**Discussion by RealNoArms** [gruntwork-io/knowledge-base#638] [How do I]
Noob looking at Terraform and Terragrunt for our IaC solution. We are managing Azure Databricks workspaces and we would like to keep config for Databricks workflow jobs per application project rather than one project for all Databricks workflows. The project teams want to keep all of their code within their own project repos. This would look something like the folder structures below (very simplified). The thought is that infrastructure will manage the networking backbone, the databricks admin team will manage the Databricks workspaces, clusters, etc. and the project dev teams will manage the code and workflows. The project workflows will be combined into a master state for deployment/syncing.  

Is there a way to solve this elegantly with Terraform and Terragrunt?

Infrastructure
     |--networking.tf

DatabricksArchitecture
     |--workspaces.tf
     |--clusters.tf
     |--repos.tf
     |--permissions.tf
     |--workflows-master.tf

ProjectA
     |--notebooks
     |        |--notebook1.py
     |--workflows
     |        |--job1.tf
     |        |--job2.tf

ProjectB
     |--notebooks
     |         |--notebook2.py
     |--workflows
     |         |--job3.tf
     |         |--job4.tf

---

<ins datetime="2023-01-18T18:43:30Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109809">Tracked in ticket #109809</a></p>
</ins>

**Comment by denis256** [comment]
Hi,
if each project's source code is only terraform code, in `terragrunt.hcl` can be defined `terraform` block referencing each repository and passing required inputs which can be common and loaded from parent directories.

https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#terraform
